packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -624,20 +443,28 @@ class HttpRouteSpec extends RouteSpec {
   }
 
   public bind(scope: Construct): RouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+    const pathMatchConfig = this.match?.path?.bind(scope).requestMatch;
+
+    // Set prefix to '/' if none on match properties are defined.
+    const prefix = areMatchPropertiesUndefined(this.match)
+      ? '/'
+      : pathMatchConfig?.prefix;
+
+    if (prefix && prefix[0] != '/') {
+      throw new Error(`Prefix Path must start with \'/\', got: ${prefix}`);
     }
 
     const httpConfig: CfnRoute.HttpRouteProperty = {
       action: {
         weightedTargets: renderWeightedTargets(this.weightedTargets),
       },
       match: {
-        prefix: prefixPath,
-        headers: this.match?.headers?.map(header => header.bind(scope).httpRouteHeader),
+        prefix: prefix,
+        headers: this.match?.headers?.map(header => header.bind(scope).headerMatch),
         method: this.match?.method,
         scheme: this.match?.protocol,
+        path: pathMatchConfig?.path,
```

Comment:
Does `path` not have to start with `'/'` too? What does it mean if it doesn't?

Reply:
Yes, exact `path` must start with '/'. Will add this validation.